1/backend/src/index.js

```diff
@@ -0,0 +1,25 @@
+const { ApolloServer } = require('apollo-server');
+const jwt = require('jsonwebtoken');
+const fs = require('fs');
+
+var SECRET_KEY = fs.readFileSync('./src/key/secret.key', 'utf8');
+
+const typeDefs = require('./schema');
+const resolvers = require('./resolvers')
+
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  context: ({ req}) => {
+    req_header = req.headers.authorization;
+    if (req_header){
+      return {token: jwt.verify(req_header, SECRET_KEY)}
+    }
```

Comment:
Is the return value of this function `{}` empty object if no request header was set? :thinking: I'm not sure but I think that would be preferable:
```js
const token = jwt.verify(req_header, SECRET_KEY) // returns null if no header is set
return { token }
```